packages/excalidraw/components/LaserTool/LaserTool.tsx

```diff
@@ -1,23 +1,35 @@
-import { useEffect, useRef } from "react";
+import { useEffect, useMemo, useRef } from "react";
 import { LaserPathManager } from "./LaserPathManager";
 import "./LaserToolOverlay.scss";
+import { AnimatedTrail } from "../../animated-trail";
 
 type LaserToolOverlayProps = {
   manager: LaserPathManager;
+  trails: AnimatedTrail[];
 };
 
-export const LaserToolOverlay = ({ manager }: LaserToolOverlayProps) => {
+export const LaserToolOverlay = ({
+  manager,
+  trails,
+}: LaserToolOverlayProps) => {
   const svgRef = useRef<SVGSVGElement | null>(null);
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const memoTrails = useMemo(() => trails, trails);
```

Comment:
what's the difference between this and passing `trails` directly as a dep to the useEffect below where it's used?

Reply:
to be honest I can just destructure the array into the deps, but passing it directly wouldn't work, right? AFAIK dep comparison is shallow